bigframes/core/expression_factoring.py

```diff
@@ -101,14 +158,26 @@ class FactoredExpression:
     sub_exprs: Tuple[nodes.ColumnDef, ...]
 
 
-def fragmentize_expression(root: nodes.ColumnDef) -> Sequence[nodes.ColumnDef]:
+def fragmentize_expression(
+    roots: Sequence[nodes.ColumnDef],
+) -> Sequence[nodes.ColumnDef]:
     """
     The goal of this functions is to factor out an expression into multiple sub-expressions.
     """
-
-    factored_expr = root.expression.reduce_up(gather_fragments)
-    root_expr = nodes.ColumnDef(factored_expr.root_expr, root.id)
-    return (root_expr, *factored_expr.sub_exprs)
+    # TODO: Fragmentize a bit less aggressively
```

Comment:
nit: Let's add that bug number after TODO so that we will know in the future what we were doing: go/todo-style#todo-style